lib/apps/responses/response.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'common/client/concerns/service_status'
+require 'common/models/base'
+
+module Apps
+  module Responses
+    ##
+    # Model for Apps responses. Body is passed straight through from the service
+    # with a validation check that it matches the expected schema.
+    #
+    # @!attribute body
+    #   @return [Hash] Validated response body.
+    # @!attribute status
+    #   @return [Integer] The HTTP status code.
+    #
+    class Response < Common::Base
+      attribute :body, String
+      attribute :status, Integer
+
+      def initialize(status, body, schema_name)
+        self.body = if status == 204
+                      nil
+                    else
+                      json_format_is_valid?(body, schema_name) ? body : {}
+                    end
+        self.status = status
```

Comment:
I am open to recommendations on how to refactor this if statement here. The only time it would not need to validate the json is when the head is a `:no_content`. Using a ternary statement versus a nested if seemed like the better of two bad options on line 25.